packages/dds/tree/src/shared-tree-core/defaultResubmitMachine.ts

```diff
@@ -75,7 +78,7 @@ export class DefaultResubmitMachine<TChange> implements ResubmitMachine<TChange>
 			// Roll back the checkout to the state before the oldest commit
 			for (let iCommit = toResubmit.length - 1; iCommit >= 0; iCommit -= 1) {
 				const commit = toResubmit[iCommit] ?? oob();
-				const rollback = this.inverter(commit, true);
+				const rollback = this.inverter(commit, true, revision);
```

Comment:
Instead of taking a single `revision`, this method needs to take a `mintRevisionTag` function so that it can make a new revision for each inverted commit in this loop.